src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_reductions_emitters.cpp

```diff
@@ -0,0 +1,89 @@
+
+size_t jit_reduce_max_emitter::get_inputs_count() const {
+    return 1;
+}
+
+void jit_reduce_max_emitter::emit_impl(const std::vector<size_t>& in_vec_idxs,
+                                       const std::vector<size_t>& out_vec_idxs) const {
+    if (host_isa_ == dnnl::impl::cpu::aarch64::asimd) {
+        emit_isa<dnnl::impl::cpu::aarch64::asimd>(in_vec_idxs, out_vec_idxs);
+    } else {
+        OV_CPU_JIT_EMITTER_THROW("Can't create jit eltwise kernel");
+    }
+}
+
+template <dnnl::impl::cpu::aarch64::cpu_isa_t isa>
+void jit_reduce_max_emitter::emit_isa(const std::vector<size_t>& in_vec_idxs,
+                                      const std::vector<size_t>& out_vec_idxs) const {
+    h->fmaxv(Xbyak_aarch64::SReg(out_vec_idxs[0]), Xbyak_aarch64::VReg4S(in_vec_idxs[0]));
+}
+
+std::set<std::vector<element::Type>> jit_reduce_max_emitter::get_supported_precisions(
+    [[maybe_unused]] const std::shared_ptr<ov::Node>& node) {
+    return {{element::f32}};
+}
+
+/// reduce_sum ///
+jit_reduce_sum_emitter::jit_reduce_sum_emitter(dnnl::impl::cpu::aarch64::jit_generator* host,
+                                               dnnl::impl::cpu::aarch64::cpu_isa_t host_isa,
+                                               const std::shared_ptr<ov::Node>& node)
+    : jit_emitter(host, host_isa, node, get_arithmetic_binary_exec_precision(node)) {}
+
+jit_reduce_sum_emitter::jit_reduce_sum_emitter(dnnl::impl::cpu::aarch64::jit_generator* host,
+                                               dnnl::impl::cpu::aarch64::cpu_isa_t host_isa,
+                                               const ov::element::Type exec_prc)
+    : jit_emitter(host, host_isa, exec_prc) {}
+
+size_t jit_reduce_sum_emitter::get_inputs_count() const {
+    return 1;
+}
+
+void jit_reduce_sum_emitter::emit_impl(const std::vector<size_t>& in_vec_idxs,
+                                       const std::vector<size_t>& out_vec_idxs) const {
+    if (host_isa_ == dnnl::impl::cpu::aarch64::asimd) {
+        emit_isa<dnnl::impl::cpu::aarch64::asimd>(in_vec_idxs, out_vec_idxs);
+    } else {
+        OV_CPU_JIT_EMITTER_THROW("Can't create jit eltwise kernel");
+    }
+}
+
+template <dnnl::impl::cpu::aarch64::cpu_isa_t isa>
+void jit_reduce_sum_emitter::emit_isa(const std::vector<size_t>& in_vec_idxs,
+                                      const std::vector<size_t>& out_vec_idxs) const {
+    const uint32_t in = in_vec_idxs[0];
+    const uint32_t out = out_vec_idxs[0];
+
+    h->orr(Xbyak_aarch64::VReg16B(out), Xbyak_aarch64::VReg16B(in), Xbyak_aarch64::VReg16B(in));
+    h->faddp(Xbyak_aarch64::VReg4S(out), Xbyak_aarch64::VReg4S(out), Xbyak_aarch64::VReg4S(out));
+    h->faddp(Xbyak_aarch64::VReg2S(out), Xbyak_aarch64::VReg2S(out), Xbyak_aarch64::VReg2S(out));
+    Xbyak_aarch64::WReg tmp = Xbyak_aarch64::WReg(h->X_TMP_0.getIdx());
+    h->umov(Xbyak_aarch64::WReg(tmp), Xbyak_aarch64::VRegSElem(out, 0, /*lane=*/0));
+    h->fmov(Xbyak_aarch64::SReg(out), Xbyak_aarch64::WReg(tmp));
```

Comment:
The current impl looks so complicated and I believe we can improve it 🤔

At the moment we use xor,  `faddp` twise and movings. In general, these emitters should contain the result in first element of vector register.  I see that the instruction  `faddp` should add elements pairwise. Can we call `faddp` just twice? Looks like this is enough to find the sum of elements of vector. Otherwise, could you please elaborate your solution?